src/main/java/com/enterprise/redcord/RedCordController.java

```diff
@@ -153,13 +153,13 @@ public String searchEntry(@RequestParam(value="searchEntry", required = false, d
             model.addAttribute("messages", messages);
             return "searchedMessages";
         } catch (ExecutionException e) {
-            e.printStackTrace();
+            logger.error("Error in searchMessageEntry method: " + e.getMessage());
             return "error";
         } catch (InterruptedException e) {
-            e.printStackTrace();
+            logger.error("Error in searchMessageEntry method: " + e.getMessage());
             return "error";
         } catch (IOException e) {
-            e.printStackTrace();
+            logger.error("Error in searchMessageEntry method: " + e.getMessage());
```

Comment:
Throwable.printStackTrace(…) should never be called because of following reasons:

1.Difficult to Retrieve Logs for Debugging:The logs written using printStackTrace is written to System.err which is hard to route or filter elsewhere. Instead, using Loggers, it is easy to retrieve logs for debugging purpose.

2. Violation of Coding Best Practices: Generally, as per coding guidelines in production-ready applications, developers need to use Logger methods for logging different level of information.

Source : https://dzone.com/articles/java-top-5-exception-handling

Reply:
Yes, this is wonderful.